front/app/components/EventAttendanceButton/index.tsx

```diff
@@ -63,9 +63,16 @@ const EventAttendanceButton = ({ event }: EventAttendanceButtonProps) => {
   const { data: phases } = usePhases(project?.data.id);
   const currentPhase = getCurrentPhase(phases?.data);
 
-  const handleEventAttendanceEvent = useCallback(() => {
-    setConfirmationModalVisible(true);
-  }, []);
+  const handleEventAttendanceEvent = useCallback(
+    (eventEmitterEvent) => {
+      const eventId = eventEmitterEvent.eventValue;
+
+      if (eventId === event.id) {
+        setConfirmationModalVisible(true);
+      }
+    },
+    [event.id]
```

Comment:
We are rendering one button and one modal for each event card. So before this change, we were opening all modals that were rendered to the screen. You would then only see the last modal, i.e. corresponding to the last event in the list. 

After this change, we only open the modal corresponding to the right event.